test/Microsoft.Azure.WebJobs.Host.EndToEndTests/DynamicConcurrencyEndToEndTests.cs

```diff
@@ -702,6 +704,15 @@ public void OnTimer(object state)
                             // don't let background exceptions propagate
                         }
                     }
+
+                    private async Task ExecuteFunctionAsync(TestEvent evt)
+                    {
+                        Interlocked.Increment(ref _outstandingInvocationCount);
+
+                        await _executor.TryExecuteAsync(new TriggeredFunctionData { TriggerValue = evt }, CancellationToken.None);
+
+                        Interlocked.Decrement(ref _outstandingInvocationCount);
```

Comment:
Should this go in a `finally` block in case `TryExecuteAsync` throws for some unexpected reason?

Reply:
Yes - this is test code so I was loose, but you're right.